lib/sycamore/sycamore/utils/cache.py

```diff
@@ -16,10 +18,24 @@ def set(self, hash_key: str, hash_value):
         pass
 
     @staticmethod
-    def get_hash_key(data: bytes) -> str:
+    def get_hash_bytes(data: bytes) -> hashlib.sha256:
         hash_sha256 = hashlib.sha256()
         hash_sha256.update(data)
-        return hash_sha256.hexdigest()
+        return hash_sha256
+
+    @staticmethod
+    def get_hash_file(file_path: str) -> hashlib.sha256:
+        sha = hashlib.sha256()
+        with open(file_path, "rb") as file:
+            file_buffer = file.read(BLOCK_SIZE)
+            while len(file_buffer) > 0:
+                sha.update(file_buffer)
+                file_buffer = file.read(BLOCK_SIZE)
+        return sha
+
+    @staticmethod
+    def update_hash(existing_hash: hashlib.sha256, new_content: Any) -> hashlib.sha256:
```

Comment:
It's unclear to me what the value of this function is?  Given it's signature, it needs to be passed a SHA-256 context.  There's no abstraction.  Why shouldn't a caller just call hash_ctx.update(buf) themselves?